packages/vscode/src/binaryValidator.ts

```diff
@@ -0,0 +1,64 @@
+import { ExtensionContext, window } from 'vscode'
+
+export const PRISMA_ALLOWED_BINS_KEY = 'prismaAllowedBins'
+
+export type BinaryState = {
+  allowed: boolean
+}
+
+export enum BinaryPopupStatus {
+  Allow,
+  Deny,
+}
+
+export type BinaryStorage = { libs: { [key: string]: BinaryState } }
+
+export const checkAndAskForBinaryExecution = async (
+  context: ExtensionContext,
+  binaryPath: string | undefined,
+  bins: BinaryStorage | undefined,
+): Promise<boolean> => {
+  // empty binary path means we are using the default binary
+  if (!binaryPath || binaryPath === '') {
+    return true
+  }
+  const existingBinaryState = bins?.libs[binaryPath]
+  if (!existingBinaryState) {
+    const item = await window.showInformationMessage(
+      `Your VS Code workspace is defining a custom binary path for the Prisma extension: "${binaryPath}" Do you want to trust this binary and let the extension execute it?`,
+      {
+        modal: true,
+      },
+      { title: 'Allow', value: BinaryPopupStatus.Allow },
+      { title: 'Deny', value: BinaryPopupStatus.Deny },
+    )
+    if (item === undefined) {
+      // dialog was cancelled
+      return false
+    }
+
+    let result = false
+
+    if (item.value === BinaryPopupStatus.Allow) {
+      result = true
+    }
+
+    if (item.value === BinaryPopupStatus.Deny) {
+      result = false
+    }
+
+    // store the state
+    bins = bins ? bins : { libs: {} }
+    bins.libs[binaryPath] = { allowed: result }
+    await context.globalState.update(PRISMA_ALLOWED_BINS_KEY, bins)
+    return result
+  } else {
+    return existingBinaryState.allowed
+  }
+}
+
+export const printBinaryCheckWarning = async () => {
+  await window.showWarningMessage(
+    "Prisma extension's binary defined in settings was not approved. Please either change the binary path or reset the approval by opening the VS Code command palette and search for 'Reset Binary execution decisions', then press Enter.",
```

Comment:
Updating the path will automatically reset the decision?

Reply:
Decision is saved based on binary location so it will not reset the decision for that binary but it will re ask for the new binary which has different file name. You can always reset all using the command menu option. 

Maybe I should also add an option to reset the current one as well. Not just all.

Reply:
Addressed via https://github.com/prisma/language-tools/pull/750/commits/654349bde801225c897d10c980bdba5048ce8797

We have now `prisma.resetCurrentFmtBinDecision` which prompts to reset decision for current bin.